File: src/main/scala/uk/yermak/audiobookconverter/fx/bind/UIUtils.scala
package uk.yermak.audiobookconverter.fx.bind

import java.io.File
import java.lang.invoke.MethodHandles

import javafx.stage.FileChooser
import org.slf4j.{Logger, LoggerFactory}
import uk.yermak.audiobookconverter.AppProperties
import uk.yermak.audiobookconverter.fx.ConverterApplication

object UIUtils {
  private val logger: Logger = LoggerFactory.getLogger(MethodHandles.lookup.lookupClass)


  def createFileChooser(title: String) = {
    val fileChooser = new FileChooser
    val sourceFolder = AppProperties.getProperty("source.folder")
    fileChooser.setInitialDirectory(getInitialDirecotory(sourceFolder))
    fileChooser.setTitle(title)
    fileChooser.getExtensionFilters.addAll(
      new FileChooser.ExtensionFilter("jpg", "*.jpg", "*.jpeg", "*.jfif"),
      new FileChooser.ExtensionFilter("png", "*.png"),
      new FileChooser.ExtensionFilter("bmp", "*.bmp"))
    val file = fileChooser.showOpenDialog(ConverterApplication.getEnv.getWindow)
    logger.debug("Opened dialog in folder: {}", sourceFolder)
    if (file != null) {
      logger.debug("Opened dialog in folder: {}", sourceFolder)
      file
    }
  }


  def getInitialDirecotory(sourceFolder: String): File = {
    if (sourceFolder == null) return new File(System.getProperty("user.home"))
    val file = new File(sourceFolder)
    if (file.exists) file
    else getInitialDirecotory(file.getParent)
  }

}

File: src/main/java/uk/yermak/audiobookconverter/fx/UIUtils.java
package uk.yermak.audiobookconverter.fx;

import javafx.stage.FileChooser;
import javafx.stage.FileChooser.ExtensionFilter;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import uk.yermak.audiobookconverter.AppProperties;

import java.io.File;
import java.lang.invoke.MethodHandles;

public final class UIUtils {
    private final Logger logger;


    private Logger logger() {
        return this.logger;
    }

    public Object createFileChooser(final String title) {
        FileChooser fileChooser = new FileChooser();
        String sourceFolder = AppProperties.getProperty("source.folder");
        fileChooser.setInitialDirectory(this.getInitialDirecotory(sourceFolder));
        fileChooser.setTitle(title);
        fileChooser.getExtensionFilters().addAll(
                new ExtensionFilter("jpg",
                        "*.jpg", "*.jpeg", "*.jfif"),
                new ExtensionFilter("png", "*.png"),
                new ExtensionFilter("bmp", "*.bmp"));
        File file = fileChooser.showOpenDialog(ConverterApplication.getEnv().getWindow());
        this.logger().debug("Opened dialog in folder: {}", new Object[]{sourceFolder});
        return file;
    }

    public File getInitialDirecotory(String sourceFolder) {
        while (sourceFolder != null) {
            File file = new File(sourceFolder);
            if (file.exists()) {
                return file;
            } else {
                sourceFolder = file.getParent();
            }

        }

        return new File(System.getProperty("user.home"));
    }

    private UIUtils() {
        this.logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
    }
}

        